### PR TITLE
Add DrupalCI chromedriver setup, with headless Chrome and Behat

### PR DESCRIPTION
**Pull request by heddn**
Thanks to @mglaman for writing up a wonderful blog post that inspired all of this.

### CODE REVIEW

File: docker-compose-services/chromedriver/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Chrome Driver
+
+This recipe adds a Chrome Driver container to a project.
```
Comment (mglaman) [inline docker-compose-services/chromedriver/README.md:3]: Technically it's ChromeDriver 🙊

File: docker-compose-services/chromedriver/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+This will allow you to have Drupal's Functional Javascript tests, among other things.
```
Comment (mglaman) [inline docker-compose-services/chromedriver/README.md:5]: We should use configuring Drupal's FunctionalJavascript test suite as an example, but in reality, I also used this for Behat with WebDriver support. The possibility is the sky's limit.

File: docker-compose-services/chromedriver/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Configuration
+
+None
+
+## Installation
+
+* Copy `docker-compose.chromedriver.yaml` to the `.ddev` folder of your project.
+* Copy (merge variables if one already exists) `docker-compose.environment.yaml` to the `.ddev` folder of your project.
+* Verify you phpunit.xml does not define environment variable for `MINK_DRIVER_ARGS_WEBDRIVER`. These should come from environment variable.
```
Comment (mglaman) [inline docker-compose-services/chromedriver/README.md:19]: s/you/your

File: docker-compose-services/chromedriver/phpunit.xml.dist [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mglaman) [inline docker-compose-services/chromedriver/phpunit.xml.dist:1]: This is this required?

Comment (heddn) [reply docker-compose-services/chromedriver/phpunit.xml.dist:1]: Not really, its just an example of how you might want to empty out your phphunit.xml.dist for functional javascript. But it also might be confusing. I'm happy to remove it.

Comment (heddn) [reply docker-compose-services/chromedriver/phpunit.xml.dist:1]: Do we want to provide any behat setup examples or not?

File: docker-compose-services/chromedriver/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+None
+
+## Installation
+
+* Copy `docker-compose.chromedriver.yaml` to the `.ddev` folder of your project.
+* Set environment variables as appropriate. Examples are for Drupal Functional and FunctionalJavascript testing.
+  * To do this, copy (merge variables if one already exists) `docker-compose.environment.yaml` to the `.ddev` folder of your project.
+* Verify your phpunit.xml does not define environment variable for `MINK_DRIVER_ARGS_WEBDRIVER`. These should come from environment variable.
+* Start (or restart) DDEV to have the service initialized: `ddev start`
```
Comment (rfay) [inline docker-compose-services/chromedriver/README.md:22]: Please add a simple test technique that people can use to verify whether it works or does not work, thanks.

**Comment by isholgueras** [comment]
Hi @heddn ,

I have one question. Maybe I'm missing something, but what is the difference between this chromedrive and the headless-chrome I contributed a few months ago [here](https://github.com/drud/ddev-contrib/tree/master/docker-compose-services/headless-chrome)?

My idea was to provide a quick access and quick example for running Behat to anyone who wanted to use chrome (or selenium + chrome) and Behat + DrupalExtension.

I've reviewed your PR but the only difference I can see is that you specifically support FunctionalJavascript tests and I didn't test it. I guess it should work but not sure :P.

If I'm know wrong, the latest version of [drupalci/chromedriver](https://hub.docker.com/layers/drupalci/chromedriver/production/images/sha256-156e26701446857450e07cfd6d48b4987b60d6f3bc3698026c844a102003b25e?context=explore) doesn't disable some features that can break some changes as I'm doing in [this custom image](https://github.com/isholgueras/chrome-headless/blob/master/Dockerfile#L27-L63) I've made.

It would be amazing if you could suggest some changes to headless-chrome or let me know what you are missing from this container to be able to have only one chrome container (to help ddev users to pick one) but as much complete as possible.

It would be good too if you can share a behat.yml or a test.feature to be able to understand what is missing in chrome-headless or what is different here.

Thanks!

**Comment by heddn** [comment]
> Hi @heddn ,
> 
> I have one question. Maybe I'm missing something, but what is the difference between this chromedrive and the headless-chrome I contributed a few months ago [here](https://github.com/drud/ddev-contrib/tree/master/docker-compose-services/headless-chrome)?


I prefer to use drupalci's headless chrome, because then I can reproduce any issues that Drupal CI also might have. Warts and all. I'm not using behat here. I'm running drupal's Functional Javascript tests, only. I have no disrespect for your solution. Just a preference to use the same docker setup as drupalci uses.

> It would be good too if you can share a behat.yml or a test.feature to be able to understand what is missing in chrome-headless or what is different here.

I'm not using Behat, so sadly that part would actually need to be added to this example.

**Comment by rfay** [comment]
It would be great to either consolidate or differentiate these two great contributions. People have these questions so often and it's so very hard for them to sort out.

**Comment by isholgueras** [comment]
Uhm, that makes sense. Thanks!

I've started the isholgueras/headless-chrome because I found some issues with different chrome images, but if you find drupalci's one reliable I can integrate it into the headless-chrome recipe and support both Functional Javascript and behat in the same recipe.

Let me find some time to add a phpunit example with drupalci's image this weekend and see if we can put this together.

**Comment by naresh-qed42** [comment]
`docker-compose.chromedriver.yml` should have the extension `.yaml` No?

**Comment by rfay** [comment]
yaml is preferred. yml didn't work in a couple of versions but I think it was restored in v1.14.2

File: docker-compose-services/chromedriver/phpunit.xml.dist (hunk 4 above)
Comment (heddn) [reply docker-compose-services/chromedriver/phpunit.xml.dist]: None have surfaced.

File: docker-compose-services/chromedriver/README.md (hunk 5 above)
Comment (heddn) [reply docker-compose-services/chromedriver/README.md]: Done

**Comment by rfay** [comment]
I rebased this (so please reset to your fork) 

I added some minor edits and combined the two yaml files.

However, I can't get the sample test to work. I see errors like this, complaining about mink session:

```
rfay@d8php2-web:/var/www/html/web$ phpunit --verbose -c core/phpunit.xml.dist core/modules/system/tests/src/FunctionalJavascript/System/DateFormatTest.php
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.20-1+0~20200710.65+debian10~1.gbpc9cbeb
Configuration: /var/www/html/web/core/phpunit.xml.dist

Testing Drupal\Tests\system\FunctionalJavascript\System\DateFormatTest
S                                                                   1 / 1 (100%)

Time: 1.33 minutes, Memory: 4.00MB

There was 1 skipped test:

1) Drupal\Tests\system\FunctionalJavascript\System\DateFormatTest::testDateFormatXss
An unexpected error occurred while starting Mink: no such session
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 4.19.76-linuxkit x86_64)

/var/www/html/web/core/tests/Drupal/FunctionalJavascriptTests/WebDriverTestBase.php:71
/var/www/html/web/core/tests/Drupal/Tests/BrowserTestBase.php:409
/var/www/html/web/core/modules/system/tests/src/FunctionalJavascript/System/DateFormatTest.php:29

OK, but incomplete, skipped, or risky tests!
Tests: 1, Assertions: 1, Skipped: 1.
rfay@d8php2-web:/var/www/html/web$ phpunit --verbose -c core/phpunit.xml.dist core/modules/system/tests/src/FunctionalJavascript/System/DateFormatTest.php
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.20-1+0~20200710.65+debian10~1.gbpc9cbeb
Configuration: /var/www/html/web/core/phpunit.xml.dist

Testing Drupal\Tests\system\FunctionalJavascript\System\DateFormatTest
E                                                                   1 / 1 (100%)

Time: 1.16 minutes, Memory: 4.00MB

There was 1 error:

1) Drupal\Tests\system\FunctionalJavascript\System\DateFormatTest::testDateFormatXss
WebDriver\Exception\NoSuchDriver: no such session
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 4.19.76-linuxkit x86_64)

/var/www/html/vendor/instaclick/php-webdriver/lib/WebDriver/Exception.php:155
/var/www/html/vendor/instaclick/php-webdriver/lib/WebDriver/AbstractWebDriver.php:157
/var/www/html/vendor/instaclick/php-webdriver/lib/WebDriver/Session.php:129
/var/www/html/vendor/behat/mink-selenium2-driver/src/Selenium2Driver.php:380
/var/www/html/vendor/behat/mink/src/Session.php:148
/var/www/html/web/core/tests/Drupal/Tests/UiHelperTrait.php:327
/var/www/html/web/core/tests/Drupal/Tests/UiHelperTrait.php:246
/var/www/html/web/core/modules/system/tests/src/FunctionalJavascript/System/DateFormatTest.php:33

ERRORS!
Tests: 1, Assertions: 5, Errors: 1.
```

**Review by rfay — APPROVED** [review]

**Pull request merged**